src/CyclicContainer.h

```diff
@@ -0,0 +1,133 @@
+/*
+  Copyright (C) 2020 SUSE LLC
+  This library is free software; you can redistribute it and/or modify
+  it under the terms of the GNU Lesser General Public License as
+  published by the Free Software Foundation; either version 2.1 of the
+  License, or (at your option) version 3.0 of the License. This library
+  is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+  License for more details. You should have received a copy of the GNU
+  Lesser General Public License along with this library; if not, write
+  to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+  Floor, Boston, MA 02110-1301 USA
+*/
+
+
+#ifndef CyclicContainer_h
+#define CyclicContainer_h
+
+#include <algorithm>
+#include <iterator>
+#include <vector>
+
+template <class T>
+class CyclicContainer
```

Comment:
So this is a container with wrap-around behaviour for `next()` and `previous()`, right?
Please add some lines of documentation to describe this.

Does it also have something like `selectFirst()` and `selectLast()` ?

Reply:
Please run `make docs` and make sure that the HTML docs make sense for the new code. A one line description for a new class is a bare minimum!

Reply:
Selecting *first* and *last* is not explicitly needed. Do you think we need to add it?